src/main/java/it/gov/pagopa/pu/sil/connector/organization/service/TaxonomyServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package it.gov.pagopa.pu.sil.connector.organization.service;
+
+import it.gov.pagopa.pu.organization.dto.generated.Taxonomy;
+import it.gov.pagopa.pu.sil.connector.organization.client.TaxonomySearchClient;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@CacheConfig(cacheNames = it.gov.pagopa.pu.sil.config.CacheConfig.Fields.taxonomy)
+public class TaxonomyServiceImpl implements TaxonomyService {
+
+  private final TaxonomySearchClient taxonomySearchClient;
+
+  public TaxonomyServiceImpl(TaxonomySearchClient taxonomySearchClient) {
+    this.taxonomySearchClient = taxonomySearchClient;
+  }
+
+  @Override
+  @Cacheable(key = "#taxonomyCode", unless = "#result == null")
```

Comment:
this cannot be null...
thus the #result == null would not be effective...
could i t be replaced into `#result.isEmpty()` ?